ironic-deployment/components/basic-auth/auth.yaml

```diff
@@ -8,30 +8,20 @@ spec:
       containers:
       - name: ironic
         volumeMounts:
-        - name: ironic-auth-config
-          mountPath: "/auth/ironic"
-          readOnly: true
         - name: ironic-inspector-auth-config
           mountPath: "/auth/ironic-inspector"
           readOnly: true
         envFrom:
         - configMapRef:
             name: ironic-htpasswd
-        - configMapRef:
-            name: ironic-inspector-htpasswd
         - configMapRef:
             name: ironic-bmo-configmap
       - name: ironic-inspector
         volumeMounts:
         - name: ironic-auth-config
           mountPath: "/auth/ironic"
           readOnly: true
-        - name: ironic-inspector-auth-config
-          mountPath: "/auth/ironic-inspector"
-          readOnly: true
```

Comment:
Please check the order, should it be that we might need to keep this here and remove the other one?

Reply:
I got confused myself, but this is actually correct. Added comments now to explain.
It goes like this: Inspector needs the credentials for accessing ironic, so it has the `ironic-auth-config`. This matches the `ironic-htpasswd` that Ironic has. And the other way: Ironic need the credentials for accessing inspector, so it has the `ironic-inspector-auth-config`. This matches the `ironic-inspector-htpasswd`.